### ISSUE TITLE
GraalVM Release 22.2.0: Exception in thread "daemon-listener" java.lang.NoClassDefFoundError: jdk.net.ExtendedSocketOptions

### ISSUE THREAD

**Issue by morph027** [AsamK/signal-cli#987]
0.10.9 with GraalVM 22.2.0

```
INFO  DaemonCommand - Starting daemon in multi-account mode
INFO  IOUtils - Listening on socket: localhost/127.0.0.1:7583
Exception in thread "daemon-listener" java.lang.NoClassDefFoundError: jdk.net.ExtendedSocketOptions
        at org.asamk.signal.util.IOUtils.getUnixDomainPrincipal(IOUtils.java:127)
        at org.asamk.signal.commands.DaemonCommand.lambda$runSocket$6(DaemonCommand.java:256)
        at java.lang.Thread.run(Thread.java:833)
        at com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:705)
        at com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:202)
```

22.1.0 works. [Release Notes](https://www.graalvm.org/release-notes/22_2/) does not really mention anything being dropped.

**Comment by AsamK** [comment]
It seems to work on my machine, on ArchLinux with `GraalVM 22.2.0 Java 17 CE (Java Version 17.0.4+8-jvmci-22.2-b06)`, running `build/native/nativeCompile/signal-cli daemon --tcp --socket`:
> INFO  DaemonCommand - Starting daemon in multi-account mode
INFO  IOUtils - Listening on socket: /run/user/1000/signal-cli/socket
INFO  IOUtils - Listening on socket: localhost/127.0.0.1:7583

Which OS are you using?

**Comment by morph027** [comment]
OS running the native executable is Ubuntu 20.04 inside a LXC container ;)

OS building was Ubuntu 20.04 Docker on top of Proxmox (Debian w/ 5.15 Kernel)

**Comment by morph027** [comment]
Hehe, latest master build has a new error ;)

```
Exception in thread "daemon-listener" java.lang.NoSuchMethodError: java.lang.invoke.MethodHandle.makeConcatWithConstants(java.net.SocketAddress, jdk.net.UnixDomainPrincipal)
```

Will try outside container...

**Comment by morph027** [comment]
Ah, okay, the error does not appear while starting, it appears when a tcp client connects

**Comment by morph027** [comment]
https://user-images.githubusercontent.com/600106/181340881-8e31a309-8dbe-4947-ae32-8d255b0d8f18.mp4

**Comment by AsamK** [comment]
Thanks, I can reproduce it now and made another fix so it doesn't crash anymore now.

**Comment by morph027** [comment]
Thank you!

Still an error with latest master.

```
Exception in thread "daemon-listener" java.lang.NoSuchMethodError: java.lang.invoke.MethodHandle.makeConcatWithConstants(java.net.SocketAddress, jdk.net.UnixDomainPrincipal)
	at org.asamk.signal.commands.DaemonCommand.lambda$runSocket$6(DaemonCommand.java:256)
	at java.lang.Thread.run(Thread.java:833)
	at com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:705)
	at com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:202)
```

Testing tcp connection with

```python
import socket
with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
    sock.connect(("localhost", 7583))
```

**Comment by AsamK** [comment]
Are you sure you pulled the latest changes and recompiled?
Because at that source location there shouldn't be any UnixDomainPrincipal anymore.

**Comment by morph027** [comment]
Hm, was wondering the same after having a look at the commit.

I'm building inside docker container from scratch within /tmp each time.

Now it looks good, must have followed this issue ;)

```
root@1751156823ee:/tmp/signal-cli# git remote -v
origin  https://github.com/AsamK/signal-cli (fetch)
origin  https://github.com/AsamK/signal-cli (push)
root@1751156823ee:/tmp/signal-cli# git branch
* master
root@1751156823ee:/tmp/signal-cli# git log
commit d69b9ea394dc4ac8298139480700611316bdb444 (grafted, HEAD -> master, origin/master, origin/HEAD)
Author: AsamK <asamk@gmx.de>
Date:   Thu Jul 28 16:57:49 2022 +0200

    Update KBS enclave

    Fixes #985
root@1751156823ee:/tmp/signal-cli# build/native/nativeCompile/signal-cli daemon --tcp
INFO  DaemonCommand - Starting daemon in multi-account mode
INFO  IOUtils - Listening on socket: localhost/127.0.0.1:7583
INFO  DaemonCommand - Accepted new client connection 0: /127.0.0.1:42988 null
INFO  DaemonCommand - Connection 0 closed: /127.0.0.1:42988 null
```
